versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/DisplayDependencyUpdatesMojo.java

```diff
@@ -506,7 +506,7 @@ static void validateGAVList(List<String> gavList, int numSections, String argume
     }
 
     private Optional<Segment> calculateUpdateScope() {
-        return allowAnyUpdates
+        return allowMajorUpdates && allowMinorUpdates && allowIncrementalUpdates
```

Comment:
The similar logic is in method  `SegmentUtils#determineUnchangedSegment` in this place looks  like complicated .... but should be refactored in separated PR - if possible.